### ISSUE TITLE
Decomposition proposal

### ISSUE THREAD

**Issue by krutoo** [motdotla/dotenv#690]
Hi, i want use `dotenv` for parse and merge objects manually, as it is done under the hood of this library.

I want to extract mechanism of modifying `process.env` by parsing result.

Can we figure out how to decompose `config()` function into for example `config()` and `merge()` or `apply()`.

In this case `config()` will be looks like:

```js
function config (options) {
  // ...
  try {
    const parsed = parse(fs.readFileSync(dotenvPath, { encoding }));

    apply({ 
      target: process.env,
      source: parsed,
      options: { debug, override },
    });

    return { parsed }
  } catch (err) {
  // ...
}
```

The `apply()` function may be looks like:

```js
function apply ({ target, source, options }) {
  const { override, debug, log } = options;

  for (const key of Object.keys(source)) {
    if (!Object.prototype.hasOwnProperty.call(target, key)) {
      target[key] = source[key];
    } else {
      if (override === true) {
        target[key] = source[key];
        debug && log(`"${key}" is already defined and WAS overwritten`);
      } else {
        debug && log(`"${key}" is already defined and was NOT overwritten`);
      }
    }
  }
}
```

**Comment by motdotla** [comment]
This is a good suggestion. Putting on todos.

**Comment by deepam-kapur** [comment]
Can I work on this @motdotla ? And can you assign me this?

**Comment by motdotla** [comment]
Let's give @krutoo the chance to do so. @krutoo, I agree with this proposal. It's a great one. Want to knock it out and I'll review and maybe make some small modifications.

**Comment by deepam-kapur** [comment]
ok sure. let me know If I could help too! @motdotla &  @krutoo

**Comment by motdotla** [comment]
@deepam-kapur if we don't here from @krutoo in 10 days then go ahead and start on the PR and we'll credit him in the changelog.

**Comment by krutoo** [comment]
i am here

**Comment by krutoo** [comment]
@deepam-kapur you can do it if you want

**Comment by motdotla** [comment]
Ok, great. @deepam-kapur go ahead and work on a PR for that

**Comment by deepam-kapur** [comment]
sure thanks!

**Comment by deepam-kapur** [comment]
@motdotla I've created a PR please review and let me know if any change is required.

**Comment by deepam-kapur** [comment]
@motdotla have you got chance to review this?

**Comment by motdotla** [comment]
@deepam-kapur reviewed and merged. it will be released as part of 16.1.0 at the end of the month.

**Comment by motdotla** [comment]
Renamed to `populate`.
